plugins/mcreator-core/blockly/js/block_procedure.js

```diff
@@ -0,0 +1,119 @@
+            const depBlock = workspace.newBlock('procedure_dependency');
+            depBlock.initSvg();
+            connection.connect(depBlock.previousConnection);
+            connection = depBlock.nextConnection;
+        }
+        return containerBlock;
+    },
+
+    compose: function(containerBlock) {
+        let depBlock = containerBlock.getInputTargetBlock('STACK');
+        const connections = [];
+        while (depBlock) {
+            if (!depBlock.isInsertionMarker())
+                connections.push(depBlock.valueConnection_);
+            depBlock = depBlock.getNextBlock();
+        }
+        const names = {};
+        for (let i = 0; i < this.depCount_; i++) {
+            const connection = this.getInput('arg' + i).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1)
+                connection.disconnect();
+            else
+                names[connection] = this.getField('name' + i).getValue();
+        }
+        this.depCount_ = connections.length;
+        this.updateShape_();
+        for (let i = 0; i < this.depCount_; i++) {
+            if (Blockly.Mutator.reconnect(connections[i], this, 'arg' + i))
+                this.getField('name' + i).setValue(names[connections[i]]);
+        }
+    },
+
+    saveConnections: function(containerBlock) {
+        let depBlock = containerBlock.getInputTargetBlock('STACK');
+        let i = 0;
+        while (depBlock) {
+            if (depBlock.isInsertionMarker()) {
+                depBlock = depBlock.getNextBlock();
+                continue;
+            }
+            const input = this.getInput('arg' + i);
+            depBlock.valueConnection_ = input && input.connection.targetConnection;
+            depBlock = depBlock.getNextBlock();
+            i++;
+        }
+    },
+
+    updateShape_: function() {
+        for (let i = 0; this.getInput('arg' + i); i++)
+            this.removeInput('arg' + i);
+        for (let i = 0; i < this.depCount_; i++) {
+            this.appendValueInput('arg' + i).appendField(javabridge.t("blockly.block.call_procedure.with"))
+                .appendField(new FieldJavaName('dependency' + i), 'name' + i).appendField(javabridge.t("blockly.block.call_procedure.arg"));
+        }
+    }
+};
+Blockly.Extensions.registerMutator('procedure_dependencies_mutator', PROCEDURE_DEPENDENCIES_MUTATOR_MIXIN, null, ['procedure_dependency']);
+
+Blockly.Blocks['call_procedure'] = {
+    init: function () {
+        this.appendDummyInput()
+            .appendField(javabridge.t("blockly.block.call_procedure"))
+            .appendField(new FieldDataListSelector('procedure'), 'procedure');
+        this.setPreviousStatement(true);
+        this.setNextStatement(true);
+        this.setColour(250);
+        this.setMutator(new Blockly.Mutator(['procedure_dependency'], this));
+        this.mixin(PROCEDURE_DEPENDENCIES_MUTATOR_MIXIN);
+    }
+};
+
+Blockly.Blocks['procedure_call'] = {

```

Comment:
I'd rename these blocks to `call_procedure_mutator_container` and `call_procedure_mutator_input`, as the current names are confusing (especially `call_procedure` and `procedure_call`)